src/modules/github/components/DataList.tsx

```diff
@@ -0,0 +1,121 @@
+'use client'
+
+import { FC, useEffect, useState } from 'react'
+
+import { TGithubPullRequest } from '../types'
+
+interface DataListProps {
+  repo: string
+  pulls: TGithubPullRequest[]
+  handleSetRepoHasCheck: (repo: string, hasCheck: boolean) => void
+}
+
+export const DataList: FC<DataListProps> = ({
+  repo,
+  pulls,
+  handleSetRepoHasCheck
+}) => {
+  const [isRepoOpen, setIsRepoOpen] = useState<boolean>(false)
+
+  const [extendedPulls, setExtendedPulls] = useState(
+    pulls.map((pull) => ({
+      number: pull.number,
+      title: pull.title,
+      isChecked: false
+    }))
+  )
+  // console.log('extendedPulls', extendedPulls)
+  const hasCheck = extendedPulls.some((data) => data.isChecked === true)
+  const hasChild = extendedPulls.length > 0
+  const isRepoCheckd = hasChild
+    ? extendedPulls.every((pull) => pull.isChecked === true)
+    : false
+
+  const handleRepoCheckboxClick = () => {
+    /*
+    1. repo 체크 상태 확인
+    2. repo 체크 되어 있으면
+      2-1) pull 체크 해제
+      2-2)  repo 체크해제
+    3.  repo 체크 안되어있으면
+      3-1)  pull 전부 체크
+      3-2)  repo 체크
+    */
+
+    if (isRepoCheckd) {
+      setExtendedPulls((prev) =>
+        prev.map((pull) => ({
+          ...pull,
+          isChecked: false
+        }))
+      )
+    } else {
+      setExtendedPulls((prev) =>
+        prev.map((pull) => ({
+          ...pull,
+          isChecked: true
+        }))
+      )
+    }
+  }
+
+  const handlePullChange = (number: number, repo: string) => {
+    setExtendedPulls((prev) =>
+      prev.map((pull) => {
+        if (pull.number === number) {
+          return { ...pull, isChecked: !pull.isChecked }
+        }
+        return pull
+      })
+    )
+  }
+  const handleRepoClick = () => {
+    setIsRepoOpen((prev) => !prev)
+  }
```

Comment:
No, I don't want this feature. Remove it

Reply:
We did that together, that function do open pull list(▼). really remove it? 0_0;;

Reply:
I'm sorry my bad. Don't remove it.